File: app/src/main/java/com/example/cykashoppinglist/entity/GenericItem.java
package com.example.cykashoppinglist.entity;

import android.view.View;

import java.util.Date;

public class GenericItem implements Item {
	private final String content;

	public GenericItem(String content) {
		this.content = content;
	}

	@Override
	public String getName() {
		return this.content;
	}

	@Override
	public Date getDate() {
		return new Date();
	}

	@Override
	public Long getId() {
		return 0L;
	}

	@Override
	public View.OnClickListener getOnClickListener() {
		// Not implemented nor will be implemented
		return null;
	}
}

File: app/src/main/java/com/example/cykashoppinglist/entity/GenericItem.kt
package com.example.cykashoppinglist.entity

import android.view.View
import java.util.*

class GenericItem(private val content: String) : Item {
	override val name: String?
		get() = content
	override val date: Date?
		get() = Date()
	override val id: Long?
		get() = 0L

	override fun hasDelete(): Boolean {
		return true
	}

	// Not implemented nor will be implemented
	override val onClickListener: View.OnClickListener?
		get() =// Not implemented nor will be implemented
			null
}